Source/JavaScriptCore/wasm/WasmSlowPaths.cpp

```diff
@@ -395,7 +397,21 @@ WASM_SLOW_PATH_DECL(simd_go_straight_to_bbq_osr)
 
     dataLogLnIf(Options::verboseOSR(), *callee, ": Entered simd_go_straight_to_bbq_osr with tierUpCounter = ", callee->tierUpCounter());
 
-    RELEASE_ASSERT(jitCompileSIMDFunction(callee, instance));
+    auto result = jitCompileSIMDFunction(callee, instance);
+    if (result.has_value()) {
+        Wasm::ExceptionType exceptionType;
+        switch (result.value()) {
+        case Wasm::Plan::Error::OutOfMemory:
+            exceptionType = Wasm::ExceptionType::OutOfMemory;
+            break;
+        case Wasm::Plan::Error::Parse:
+            exceptionType = Wasm::ExceptionType::OutOfMemory;
```

Comment:
This shouldn't be possible since we already parsed the code, right?

Reply:
Oops